doc/classes/Viewport.xml

```diff
@@ -282,6 +282,12 @@
 		<member name="use_xr" type="bool" setter="set_use_xr" getter="is_using_xr" default="false">
 			If [code]true[/code], the viewport will use the primary XR interface to render XR output. When applicable this can result in a stereoscopic image and the resulting render being output to a headset.
 		</member>
+		<member name="vrs_mode" type="int" setter="set_vrs_mode" getter="get_vrs_mode" enum="Viewport.VRSMode" default="0">
+			The VRS mode that is used for this viewport. Note, if hardware does not support VRS this property is ignored.
+		</member>
+		<member name="vrs_texture" type="Texture2D" setter="set_vrs_texture" getter="get_vrs_texture">
+			Texture to use when [member vrs_mode] is set to [constant Viewport.VRS_TEXTURE].
+		</member>
```

Comment:
Since there are no project settings for this, there doesn't seem to be a way to change the VRS mode without using a script or a custom viewport. This should probably be added by allowing the user to specify the mode for the root Viewport, and a texture path. If changing the mode to XR doesn't make sense for the root Viewport, then the option should be hidden and only Disabled and Texture should be displayed.

Also, I think a global VRS override should be supported for performance testing purposes. This could be a list of premade textures with all pixels set to 1×2, 2×1, 2×2 and so on. This would also be good to expose as a VRS mode in the project settings.

Reply:
I think the biggest issue especially with VRS on XR is that you generally don't want to turn it on until it is available. With non-XR use cases the texture may not be available from the start especially if the texture is not a fixed texture but one generated (at the moment only possible for code, but we should investigate some day creating viewports based on different texture formats including R8_NORM

It's also very likely that you wouldn't turn VRS on in a menu screen, but do turn it on in the main viewport when your game actually starts (especially in non-XR use cases), that likely will only be a feasible option to do in code.

That all said, out of principle having these settings available as project settings, even if only to make it clear this functionality is supported on the main viewport, and for making it easier to run test projects, yeah it may not be a bad idea to just add them and accept they have limited uses.

Reply:
Ok I added the VRS mode and VRS texture for the main viewport to the project settings :)